lib/ex_unit/test/ex_unit/doc_test_test.exs

```diff
@@ -474,35 +481,50 @@ defmodule ExUnit.DocTestTest do
 
     assert output =~ """
              3) doctest module ExUnit.DocTestTest.Invalid (3) (ExUnit.DocTestTest.ActuallyCompiled)
+                test/ex_unit/doc_test_test.exs:#{doctest_line}
+                Doctest failed
+                code:  iex> a = "This is an egregiously long text string."
+                       iex> b = ~r{an egregiously long}
+                       iex> c = "a slightly shorter"
+                       iex> String.replace(a, b, c)
+                       "This is a much shorter text string."
+                left:  "This is a slightly shorter text string."
+                right: "This is a much shorter text string."
```

Comment:
I love this! This is exactly what I had in mind.

The only remaining question are:

  * does if `left` and `right` make sense now?
  * Should we show the last line of iex (the one without iex>) given that we show it right below anyway?

Reply:
I would like to add another question: how should it behave in case of several assertions? 

The current implementation behaves:

**Error at the start**

```elixir
# Doctest

      iex> a = 1
      iex> b = 2
      iex> a + b
      0
      iex> c = 3
      iex> a + c
      4
      iex> a + b + c
      5

# Error message

     Doctest failed
     code:  iex> a = 1
            iex> b = 2
            iex> a + b
            0
     left:  3
     right: 0
```

**Error at the middle**

```elixir
# Doctest

      iex> a = 1
      iex> b = 2
      iex> a + b
      3
      iex> c = 3
      iex> a + c
      0
      iex> a + b + c
      5

# Error message

     Doctest failed
     code:  iex> c = 3
            iex> a + c
            0
     left:  4
     right: 0
```

**Error at the end**

```elixir
# Doctest

      iex> a = 1
      iex> b = 2
      iex> a + b
      3
      iex> c = 3
      iex> a + c
      4
      iex> a + b + c
      0

# Error message

     Doctest failed
     code:  iex> a + b + c
            5
     left:  6
     right: 0
```

Reply:
@josevalim I think `left` makes sense, since it evaluates the expression and makes it easier to see the failed assertion. Instead of left and right we could also do
` assertion failed "This is a slightly shorter text string." == "This is a much shorter text string."`. With his we could potentially also omit the last line of iex.

@fertapric In the perfect world we should include all the "relevant" lines, but maybe for now we should just include all lines until the assertion error.

Reply:
> does if left and right make sense now?

I think `expected` and `actual` makes more sense than `left` and `right`.

> Should we show the last line of iex (the one without iex>) given that we show it right below anyway?

Not if we show the equivalent of `left` and `right` as well. We could show the last line and instead only print `result` but I prefer the former.